phoenicis-repository/src/main/java/org/phoenicis/repository/dto/ApplicationDTO.java

```diff
@@ -72,6 +69,27 @@ public String getDescription() {
         return scripts;
     }
 
+    /**
+     * Returns the main miniature belonging to this {@link ApplicationDTO}.
+     * The main miniature is the miniature with the file name <code>main.png</code>.
+     * If no such miniature exists the first miniature of this {@link ApplicationDTO} is returned.
+     * If this {@link ApplicationDTO} contains no miniatures {@link Optional#empty()} is returned.
+     *
+     * @return An optional with the found URI, or {@link Optional#empty()} if no miniature exists
+     */
+    public Optional<URI> getMainMiniature() {
+        Optional<URI> result = this.miniatures.stream().filter(uri -> uri.getPath().endsWith("main.png")).findFirst();
```

Comment:
A DTO cannot return an Optional type, Jackson would not serialize it properly.

In principle, the responsibility of à DTO is *just* to hold data. Any logic should be put in another place